lib/system_win32.c

```diff
@@ -104,6 +115,16 @@ CURLcode Curl_win32_init(long flags)
       Curl_if_nametoindex = pIfNameToIndex;
   }
 
+  s_ws2_32Dll = Curl_load_library(TEXT("ws2_32.dll"));
+  if(s_ws2_32Dll) {
+    *(FARPROC*)&ptrFreeAddrInfoExW = GetProcAddress(s_ws2_32Dll,
+      "FreeAddrInfoExW");
+    *(FARPROC*)&ptrGetAddrInfoExCancel = GetProcAddress(s_ws2_32Dll,
+      "GetAddrInfoExCancel");
+    *(FARPROC*)&ptrGetAddrInfoExW = GetProcAddress(s_ws2_32Dll,
+      "GetAddrInfoExW");
+  }
```

Comment:
an `if` implies it can fail. What happens if the DLL loading fails?

Reply:
then these will be NULL. When the GetAddrInfoExW is attempted to be used all these are checked in asyn-thread.c line 632

Reply:
Use Winsock only if it is already loaded. I would guard the block with USE_WINSOCK and then GetModuleHandleA("ws2_32") instead and get rid of the loadlibrary/freelibrary.

Reply:
the problem is then if the winsock gets unloaded/reloaded during runtime the pointers might end up dangling. LoadLibrary is the right approach here.